chain/network/src/peer/peer_actor.rs

```diff
@@ -1107,6 +1107,13 @@ impl Handler<QueryPeerStats> for PeerActor {
     fn handle(&mut self, msg: QueryPeerStats, _: &mut Self::Context) -> Self::Result {
         #[cfg(feature = "delay_detector")]
         let _d = DelayDetector::new("query peer stats".into());
+
+        // Truncate to fix reporting in cases there is no data read/written.
+        // TODO(#5218) Refactor this code to use `SystemTime`
+        let now = millis_since_epoch();
+        self.tracker.sent_bytes.truncate(now);
```

Comment:
I don't like this solution - as it makes the API to rate counter quite confusing ("and remember to call the 'truncate' before calling bytes_per_minute if you didn't send any bytes recently")...

could we fix it within rate_counter.rs instead (by possibly checking the last time when truncate was run and running it from there?)

Reply:
@mm-near I rewrite to change, to get and update at the same time.